### PR TITLE
Update docs to explain config option merging

### PR DESCRIPTION
**Pull request by danny-andrews**
**What is the purpose of this pull request? (put an "X" next to item)**

[X] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md))
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

**What changes did you make? (Give an overview)**

Updated the docs to explain how config options are merged from base configs.

**Is there anything you'd like reviewers to focus on?**

There was a lot of [confusion](https://github.com/eslint/eslint/issues/6361) generated by the merging behavior of rules inherited from base configs, so I thought I would clear that up here. There still may be [scenarios](https://github.com/eslint/eslint/issues/6144) which arise where options cannot be overridden from base configs, but we'll deal with those on a per-case basis.

### CODE REVIEW

**Comment by not-an-aardvark** [comment]
Thanks for the PR! I think the build is failing because markdownlint wants you to use four spaces in markdown indentation, rather than two.

File: docs/user-guide/configuring.md [hunk 1]
```diff
@@ -566,8 +566,14 @@ ESLint extends configurations recursively so a base configuration can also have
 The `rules` property can do any of the following to extend (or override) the set of rules:
 
 * enable additional rules
-* override default options for rules from base configurations
-* disable rules from base configurations
+* change an inherited rule's severity without changing its options:
+    * Base config: `"eqeqeq": [2, "allow-null"]`
```
Comment (nzakas) [inline docs/user-guide/configuring.md]: Can you use the "error", "warning", "off" notation instead of the numeric notation for severity?

**Comment by kaicataldo** [comment]
@danny-andrews Sorry to have you do this again, but mind signing the JS Foundation CLA by clicking on "Details" next to the `license/cla` check in the checks below?

File: docs/user-guide/configuring.md (hunk 1 above)
Comment (danny-andrews) [inline docs/user-guide/configuring.md]: Done.

**Comment by danny-andrews** [comment]
If this looks ready to go, I'll squash and amend the commit message.

**Comment by nzakas** [comment]
@danny-andrews don't worry, we can squash and change the commit message directly. Thanks for the contribution!

**Pull request merged**